subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c

```diff
@@ -2167,9 +2175,10 @@ static void *radio_ccm_ext_rx_pkt_set(struct ccm *cnf, uint8_t phy, uint8_t pdu_
 #if defined(CONFIG_BT_CTLR_PHY_CODED)
 #if defined(CONFIG_HAS_HW_NRF_RADIO_BLE_CODED)
 	case PHY_CODED:
-		mode |= (CCM_MODE_DATARATE_125Kbps <<
-			 CCM_MODE_DATARATE_Pos) &
+#if !defined(CONFIG_SOC_COMPATIBLE_NRF54LX)
+		mode |= (CCM_MODE_DATARATE_125Kbps << CCM_MODE_DATARATE_Pos) &
 			CCM_MODE_DATARATE_Msk;
```

Comment:
Given how `mode` is set and modified through this function, I would consider either defining the `mode` as a macro or a function that takes the PHY as input. 

The current implementation where `mode` is assigned 3 different places and then has 7 `|=` over 80 lines is hard to follow, and likely prone to errors.

Reply:
> Different SoCs have different supported bitmaps and features to be assigned

I realize that, and that was my argument to why such a function should exist - Understanding the final value of `mode` in this function is quite hard as it's assigned and updated so many places with so many different conditionals

Reply:
But since this was the code before, it's not a blocker for this PR. It does however IMO reduce readability and maintainability